### PR TITLE
Odocrypt Support

### PR DESCRIPTION
**Pull request by SmartArray**
I've disabled the hashing due to a stack smash issue, that I couldn't locate for now.

### CODE REVIEW

File: crypto/odocrypt.c [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    }
+}
+
+void OdoCrypt_ApplyRotations(uint64_t* state, const int* rotations)
+{
+    uint64_t next[STATE_SIZE];
+    
+    // rotate memcpy (rotate by one)
+    // @Mental: Please check this. Alternatively, I would achieve it with a loop.
+    // Original: std::rotate_copy(state, state+1, state+STATE_SIZE, next);
```
Comment (MentalCollatz) [inline crypto/odocrypt.c:135]: If you're going to use memcpy you have to multiply the third argument by sizeof(uint64_t).  I would just do a loop:
```
for (int i = 1; i < STATE_SIZE; i++)
    next[i-1] = state[i];
next[STATE_SIZE-1] = state[0];
```

File: crypto/odocrypt.c [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    for (int i = 0; i < STATE_SIZE; i++)
+    {
+        for (int j = 0; j < 8; j++)
+        {
+            bytes[8*i + j] = (state[i] >> (8*j)) & 0xff;
+        }
+    }
+}
+
+void OdoCrypt_ApplySboxes(uint64_t* state, const uint8_t** sbox1, const uint16_t** sbox2)
```
Comment (MentalCollatz) [inline crypto/odocrypt.c:106]: I think there's a problem trying to cast a `uint8_t[SMALL_SBOX_COUNT][1 << SMALL_SBOX_WIDTH]` to `const uint8_t**`.

File: crypto/odocrypt.c (hunk 1 above)
Comment (SmartArray) [reply crypto/odocrypt.c:135]: Yeah, I actually know that (as you can see in the rest of the source code). Just forgot it here.
I prefer the loop as well :-)

File: crypto/odocrypt.c (hunk 2 above)
Comment (SmartArray) [reply crypto/odocrypt.c:106]: Oh! That sounds good. Would you like to help me with that one? 
Also, the Odocrypt_Init-function does not do such a cast?

**Pull request merged**